File: tombnb22_ui/src/services/ReviewService.js
import http from "../http-common";

class ReviewService {
  getAllReviews(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/review/reviews`, searchDTO);
  }

  get(reviewId) {
    return this.getRequest(`/review/${reviewId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/review?field=${matchData}`, null);
  }

  addReview(data) {
    return http.post("/review/addReview", data);
  }

  update(data) {
  	return http.post("/review/updateReview", data);
  }
  
  uploadImage(data,reviewId) {
  	return http.postForm("/review/uploadImage/"+reviewId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ReviewService();

File: src/main/java/com/tombnb22/service/ReviewService.java
package com.tombnb22.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.tombnb22.domain.Review;
import com.tombnb22.dto.ReviewDTO;
import com.tombnb22.dto.ReviewSearchDTO;
import com.tombnb22.dto.ReviewPageDTO;
import com.tombnb22.dto.ReviewConvertCriteriaDTO;
import com.tombnb22.service.GenericService;
import com.tombnb22.dto.common.RequestDTO;
import com.tombnb22.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ReviewService extends GenericService<Review, Integer> {

	List<Review> findAll();

	ResultDTO addReview(ReviewDTO reviewDTO, RequestDTO requestDTO);

	ResultDTO updateReview(ReviewDTO reviewDTO, RequestDTO requestDTO);

    Page<Review> getAllReviews(Pageable pageable);

    Page<Review> getAllReviews(Specification<Review> spec, Pageable pageable);

	ResponseEntity<ReviewPageDTO> getReviews(ReviewSearchDTO reviewSearchDTO);
	
	List<ReviewDTO> convertReviewsToReviewDTOs(List<Review> reviews, ReviewConvertCriteriaDTO convertCriteria);

	ReviewDTO getReviewDTOById(Integer reviewId);







}





